### PR TITLE
Add noeval as a hook

### PR DESCRIPTION
**Pull request by brianjbuck**

### CODE REVIEW

File: .pre-commit-hooks.yaml [hunk 1]
```diff
@@ -28,3 +28,9 @@
     entry: ' `[^`]+[^_]`([^_]|$)'
     language: pygrep
     types: [rst]
+-   id: python-noeval
+    name: Check for eval()
+    description: 'A quick check for `eval()` built-in function'
+    entry: '\beval\s*\((.*|\s*\))'
+    language: pygrep
+    types: [python]
```
Comment (asottile) [inline .pre-commit-hooks.yaml]: please alphabetize

Also the regex can be much simpler:

```
\beval\(
```

(also if you want to add some tests (optional), there's a little harness in `tests/hooks_test.py` -- might be good to show that `eval("foo")` gets matched and `ast.literal_eval(bar)` does not for instance)

Comment (brianjbuck) [reply .pre-commit-hooks.yaml]: It looks like I didn't need to update the Readme as well.

Comment (asottile) [reply .pre-commit-hooks.yaml]: yeah there's a pre-commit hook which will put the appropriate text there -- `pre-commit run --all-files` will put the README into good shape

File: tests/hooks_test.py [hunk 2]
```diff
@@ -57,3 +57,23 @@ def test_python_check_blanket_noqa_positive(s):
...
+
+
+@pytest.mark.parametrize(
+    's',
+    (
+        'literal_eval()',
+    ),
+)
+def test_python_check_negative(s):
+    assert not HOOKS['python-noeval'].search(s)
```
Comment (asottile) [inline tests/hooks_test.py]: neither of these testcases are valid _calls_ of the two functions -- also if there's only one example there's no need for `parametrize`

File: .pre-commit-hooks.yaml [hunk 3]
```diff
@@ -10,6 +10,12 @@
     language: pygrep
     entry: 'assert .*\.(not_called|called_)'
     types: [python]
+-   id: python-noeval
+    name: Check for eval()
```
Comment (asottile) [inline .pre-commit-hooks.yaml]: for consistency, the rest of the hook names are lower cased

Comment (brianjbuck) [reply .pre-commit-hooks.yaml]: Actually, they aren't: https://github.com/pre-commit/pygrep-hooks/blob/master/.pre-commit-hooks.yaml#L2

But if you like I can modify that line as well. But I'm going to have to do that later this afternoon.

Comment (asottile) [reply .pre-commit-hooks.yaml]: yeah let's fix that one too, it was inconsistent and I just didn't notice it on PR :(

File: tests/hooks_test.py (hunk 2 above)
Comment (asottile) [reply tests/hooks_test.py]: the first half of this is still an issue `eval()` and `literal_eval()` aren't representative testcases of what you'd see in actual code

**Comment by mxr** [comment]
@brianjbuck are you still hacking on this? I can resolve conflicts if you want?

**Comment by brianjbuck** [comment]
@mxr, Oh, I had forgotten about this. Thanks for the follow-up. Yeah, if you could resolve the conflicts that would be great.

**Comment by mxr** [comment]
@brianjbuck I don't have access to edit this PR but I can open a new one

**Comment by mxr** [comment]
#14 and #15

**Comment by asottile** [comment]
via #15

**Pull request closed without merging**